mutiny-core/src/lib.rs

```diff
@@ -2191,6 +2197,43 @@ impl<S: MutinyStorage> MutinyWallet<S> {
     pub fn is_safe_mode(&self) -> bool {
         self.safe_mode
     }
+
+    /// Calls upon a Cashu mint and withdrawls/redeem/melts the token from it.
+    pub async fn melt_cashu_token(&self, token: TokenV3) -> Result<bool, MutinyError> {
+        // NOTE: for now it assumes that there is a single token being melted
+        let meltable_token = token.tokens.first();
+        let mint_url = meltable_token.map(|mt| mt.mint)?.unwrap(); // TODO: (@leonardo+elnosh) handle the unwrap() before moving to review + merging
+        let proofs = meltable_token.map(|mt| mt.proofs)?;
+
+        // TODO: (@leonardo+elnosh) We need a better strategy to get the optimal "meltable" amount.
+        let amount_sats = proofs.total_amount() * (1 - 1 / 100);
+        let mutiny_invoice = self
+            .create_invoice(amount_sats, vec!["Cashu Token Melt".to_string()])
+            .await?;
+        let mutiny_invoice_str = mutiny_invoice
+            .bolt11
+            .expect("The invoice should have BOLT11");
+
+        let response = self
+            .cashu_mint_client
+            .post_melt_tokens(&mint_url, proofs, mutiny_invoice_str, outputs)
+            .await;
```

Comment:
note: @elnosh mentioned that this other newer method wouldn't require a trait implementation (although the new code hasn't been released yet)
https://github.com/ngutech21/moksha/blob/25bc15d5bff77feb91f2c8d404445778d9bd4cee/moksha-wallet/src/client/crossplatform.rs#L70-L83

Reply:
Awesome to see cashu being integrated in my favorite ln-wallet. 
The released version of moksha 0.1.2 is still using the legacy api in the wallet. I will create a new release that will support v1. Since nutstash already supports v1, I think this is fine. As mentioned above I am also working on simplifying the wallet-api. If you need anything feel free to open a ticket or contact me.